src/entities.rs

```diff
@@ -340,7 +349,8 @@ impl Entities {
         if meta.generation != entity.generation {
             return Err(NoSuchEntity);
         }
-        meta.generation += 1;
+        let to_add = u32::from(meta.generation) + 1;
+        meta.generation = Entities::ret_non_zero_u32(to_add);
```

Comment:
This is unsound, because `u32::from(meta.generation) + 1` might wrap. Use `checked_add` and handle wrapping explicitly.